hack/helm/release/internal/helm/helm.go

```diff
@@ -0,0 +1,616 @@
+				return true
+			}
+			return false
+		}),
+		retry.Attempts(60),
+		retry.Delay(1*time.Minute),
+		retry.OnRetry(func(n uint, err error) {
+			log.Printf("retry #%d: %s\n", n, err)
+		}),
+	)
+
+	return nil
+}
+
+func updateHelmRepositories() error {
+	// simulates helm repo update
+	f, err := repo.LoadFile(cli.New().RepositoryConfig)
+	if err != nil {
+		return fmt.Errorf("while loading repository file: %w", err)
+	}
+	for _, r := range f.Repositories {
+		cr, err := repo.NewChartRepository(r, getter.All(cli.New()))
+		if err != nil {
+			return fmt.Errorf("while getting new chart repository: %w", err)
+		}
+		if _, err = cr.DownloadIndexFile(); err != nil {
+			return fmt.Errorf("while updating repository index for (%s): %w", cr.Config.Name, err)
+		}
+		log.Printf("Updated repository index for (%s)\n", cr.Config.Name)
+	}
+	return nil
+}
+
+// uploadCharts will perform the following:
+//  1. Ensure that the Helm "stable" repository exists in the repository list.
+//  2. For each Helm chart, copy the chart to a temporary directory
+//  3. For each Helm chart's dependency, if the dependency's repository
+//     is set to 'https://helm.elastic.co', and the flag 'charts-repo-url'
+//     is set to another repository (such as a dev helm repo), the dependency's
+//     repository will be re-written to the given 'charts-repo-url' flag.
+//  4. Run 'Helm dependency update' for the Helm Chart.
+//  5. Run 'Helm package chart' for the Helm Chart to generate a tarball.
+//  6. Copy the tarball to the GCS bucket.
+func uploadCharts(ctx context.Context, tempDir string, charts []chart, conf ReleaseConfig) error {
+	type helmRepo struct {
+		name, url string
+	}
+	for _, r := range []helmRepo{{name: "stable", url: stableHelmChartsURL}, {name: conf.Bucket, url: conf.ChartsRepoURL}} {
+		if err := addHelmRepository(r.name, r.url); err != nil {
+			return err
+		}
+	}
+	log.Printf("Done adding (%s) repository.\n", stableHelmChartsURL)
+	for _, chart := range charts {
+		chartPath := filepath.Join(tempDir, chart.Name)
+		if err := copyChartToTemporaryDirectory(chart, chartPath); err != nil {
+			return err
+		}
+		log.Printf("Attempting to add chart (%s)\n", chart.Name)
+		if err := updateDependencyChartURLs(chartPath, conf.ChartsRepoURL); err != nil {
+			return err
+		}
+		if err := runChartDependencyUpdate(chartPath, chart.Name); err != nil {
+			return err
+		}
+		if err := packageHelmChart(chartPath, chart.Name); err != nil {
+			return err
+		}
+		source := fmt.Sprintf("%s-*.tgz", chart.Name)
+		if err := copyChartToGCSBucket(ctx, copyChartToBucketConfig{
+			bucket:        conf.Bucket,
+			chartName:     chart.Name,
+			tempDirectory: tempDir,
+			sourceGlob:    source,
+			dryRun:        conf.DryRun,
+		}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// addHelmRepository will add the given Helm repository name + url
+// to the local Helm configuration.
+func addHelmRepository(name, url string) error {
```

Comment:
`name` and `url` arguments are not used. We always add "stable", "https://charts.helm.sh/stable" instead, normal?

Reply:
This was fixed.  I was initially only adding stable, and then realized that for dependencies sake, itself would be needed.  Resolved.